bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/NumberValue.java

```diff
@@ -56,14 +57,25 @@ public class NumberValue extends Value {
 
     public NumberValue(@Nullable BigDecimal min, @Nullable BigDecimal max, @Nullable BigDecimal step,
             @Nullable Unit<?> unit) {
-        super(CoreItemFactory.NUMBER, List.of(DecimalType.class, QuantityType.class, IncreaseDecreaseType.class,
+        super(getItemType(unit), List.of(DecimalType.class, QuantityType.class, IncreaseDecreaseType.class,
                 UpDownType.class, StringType.class));
         this.min = min;
         this.max = max;
         this.step = step == null ? BigDecimal.ONE : step;
         this.unit = unit;
     }
 
+    private static String getItemType(@Nullable Unit<?> unit) {
+        if (unit == null) {
+            return CoreItemFactory.NUMBER;
+        }
+        String dimension = UnitUtils.getDimensionName(unit);
```

Comment:
You need to add your special Mirek case handler here. And only call getDimensionName() for other units.